### PR TITLE
Add a no client option

### LINKED ISSUE #9708: No client option
##### **Overview of the feature request**

Since we now have a question for the front-end framework we can also propose a `No client` choice that has the same effect as `--skipClient`

##### **Motivation for or Use Case**

I think JHipster can be used for every type of applications, even monolithic ones without front-end

-   [x] Checking this box is mandatory (this is just to show you read everything)

### PR DESCRIPTION
**Pull request by cbornet**
Fix #9708

-   Please make sure the below checklist is followed for Pull Requests.

-   [x] [Travis tests](https://travis-ci.org/jhipster/generator-jhipster/pull_requests) are green
-   [x] Tests are added where necessary
-   [x] Documentation is added/updated where necessary
-   [x] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/master/CONTRIBUTING.md) are followed

### CODE REVIEW

File: generators/server/index.js [hunk 1]
```diff
@@ -439,6 +439,9 @@ module.exports = class extends BaseBlueprintGenerator {
                 if (this.configOptions.clientFramework) {
                     this.clientFramework = this.configOptions.clientFramework;
                 }
+                if (this.configOptions.skipClient) {
+                    this.skipClient = this.configOptions.skipClient;
```
Comment (murdos) [inline generators/server/index.js]: Is there any use yet? Or is this just for future use, and to be sure the information is available in this subgenerator?

Comment (cbornet) [reply generators/server/index.js]: Yes it's important as some files of the server are different if skipClient is set (pom.xml, README, ...)

Comment (murdos) [reply generators/server/index.js]: Ok, but this assignation is already done in `this.setupServerOptions(this)`: https://github.com/jhipster/generator-jhipster/blob/master/generators/server/index.js#L61

Comment (cbornet) [reply generators/server/index.js]: It needs to be done again at `configuring` phase after the `prompting` phase. You can try to remove it and you'll see the server will be generated with front-end related code.

Comment (murdos) [reply generators/server/index.js]: Ok, I see. The `prompting` phase of the `client` subgen will indeed happen between the different phases of the `server` subgen.

**Pull request merged**